plugins/scaffolder-react/api-report.md

```diff
@@ -0,0 +1,464 @@
+## API Report File for "@backstage/plugin-scaffolder-react"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+/// <reference types="react" />
+
+import { ApiHolder } from '@backstage/core-plugin-api';
+import { ApiRef } from '@backstage/core-plugin-api';
+import { Dispatch } from 'react';
+import { Extension } from '@backstage/core-plugin-api';
+import { ExternalRouteRef } from '@backstage/core-plugin-api';
+import { FieldProps } from '@rjsf/core';
+import { FieldProps as FieldProps_2 } from '@rjsf/utils';
+import { FieldValidation } from '@rjsf/core';
+import { FieldValidation as FieldValidation_2 } from '@rjsf/utils';
+import type { FormProps as FormProps_2 } from '@rjsf/core-v5';
+import { JsonObject } from '@backstage/types';
+import { JSONSchema7 } from 'json-schema';
+import { JsonValue } from '@backstage/types';
+import { Observable } from '@backstage/types';
+import { PathParams } from '@backstage/core-plugin-api';
+import { PropsWithChildren } from 'react';
+import { default as React_2 } from 'react';
+import { RouteRef } from '@backstage/core-plugin-api';
+import { SetStateAction } from 'react';
+import { SubRouteRef } from '@backstage/core-plugin-api';
+import { TaskSpec } from '@backstage/plugin-scaffolder-common';
+import { TaskStep } from '@backstage/plugin-scaffolder-common';
+import { TemplateEntityV1beta3 } from '@backstage/plugin-scaffolder-common';
+import { UIOptionsType } from '@rjsf/utils';
+import { UiSchema } from '@rjsf/utils';
+
+// @public (undocumented)
+export const actionsRouteRef: SubRouteRef<undefined>;
+
+// @alpha
+export const createFieldValidation: () => FieldValidation_2;
+
+// @alpha
+export function createNextScaffolderFieldExtension<
+  TReturnValue = unknown,
+  TInputProps extends UIOptionsType = {},
+>(
+  options: NextFieldExtensionOptions<TReturnValue, TInputProps>,
+): Extension<FieldExtensionComponent<TReturnValue, TInputProps>>;
+
+// @public
+export function createScaffolderFieldExtension<
+  TReturnValue = unknown,
+  TInputProps = unknown,
+>(
+  options: FieldExtensionOptions<TReturnValue, TInputProps>,
+): Extension<FieldExtensionComponent<TReturnValue, TInputProps>>;
+
+// @public
+export type CustomFieldExtensionSchema = {
+  returnValue: JSONSchema7;
+  uiOptions?: JSONSchema7;
+};
+
+// @public
+export type CustomFieldValidator<TFieldReturnValue> = (
+  data: TFieldReturnValue,
+  field: FieldValidation,
+  context: {
+    apiHolder: ApiHolder;
+  },
+) => void | Promise<void>;
+
+// @public (undocumented)
+export const editRouteRef: SubRouteRef<undefined>;
+
+// @alpha
+export const extractSchemaFromStep: (inputStep: JsonObject) => {
+  uiSchema: UiSchema;
+  schema: JsonObject;
+};
+
+// @public
+export const FIELD_EXTENSION_KEY = 'scaffolder.extensions.field.v1';
+
+// @public
+export const FIELD_EXTENSION_WRAPPER_KEY = 'scaffolder.extensions.wrapper.v1';
```

Comment:
Hmm - I mean it would be nice if they weren't, but they would have to be unless we exported a hook that collected the `FieldExtensions` which I guess we could do.

Reply:
We could probably just declare them in both packages too tbh